src/systems/trajectory_follower/TrajectoryFollower.cc

```diff
@@ -376,11 +387,31 @@ void TrajectoryFollower::PreUpdate(
   {
     forceWorld = (*comPose).Rot().RotateVector(
       ignition::math::Vector3d(this->dataPtr->forceToApply, 0, 0));
-  }
 
+    // force angular velocity to be zero when bearing is reached
+    if (this->dataPtr->forceZeroAngVel &&
+        math::equal(std::abs(bearing.Degree()), 0.0, 1.0))
```

Comment:
Do we need the second expression `math::equal(std::abs(bearing.Degree()), 0.0, 1.0)`?

We're supposed to be within the acceptable range bearing. If we need higher bearing accuracy, I think we should change the parameter `bearingTolerance`.

Reply:
The behavior I found without this check is that as soon as the vehicle moves forward 1 iteration, its bearing immediately becomes outside of `bearingTolerance`. So it constantly switched back and forth between rotation and moving forward. The problem was that removing the `AngularVelocityCmd` component is found to be very expensive and this locks up simulation. With this check, I still do notice a slight pause in simulation when the component is being removed but it only happens when occasionally when bearing needs to be corrected.

Reply:
I tweaked the 2nd tol check to be a percentage of `bearingTolerance`. [aa8cf68](https://github.com/ignitionrobotics/ign-gazebo/pull/1349/commits/aa8cf68ec8894ec110fa0df9c551d6d9ba755d30)